rosbag2_storage/include/rosbag2_storage/yaml.hpp

```diff
@@ -95,7 +75,15 @@ struct convert<rosbag2_storage::TopicMetadata>
     node["name"] = topic.name;
     node["type"] = topic.type;
     node["serialization_format"] = topic.serialization_format;
-    node["offered_qos_profiles"] = topic.offered_qos_profiles;
+    std::vector<rosbag2_storage::Rosbag2QoS> to_encode;
```

Comment:
Here, a conversion from `std::vector<rclcpp::QoS>` to serialized YAML (`std::string`) is needed.